### DISCUSSION TITLE
JSON for Wemos D1 mini with 3 „Sensors“

### DISCUSSION THREAD

**Discussion by Hanimaniac** [RavenSystem/esp-homekit-devices#1734] [MEPLHAA Scripts]
Hey Everyone,
I‘d like to use a wemos d1 for an input from 3 External Sensors. A Rainsensor, a Lightsensor and a Windsensor. 

i use this Json: {"c": { "l": 13,"b":[{"g":0,"t":5}]},"a": [{"t": 12,"i": 60,"0": { "r": [{ "g": 1, "v": 0 }] },"1": { "r": [{ "g": 1, "v": 1 }] },"f1": [{ "g": 16 }]},{"t": 12,"i": 60,"0": { "r": [{ "g": 1, "v": 0 }] },"1": { "r": [{ "g": 2, "v": 1 }] },"f1": [{ "g": 14 }]},{"t": 12,"i": 60,"0": { "r": [{ "g": 2, "v": 0 }] },"1": { "r": [{ "g": 2, "v": 1 }] },"f1": [{ "g": 5 }]}]}. 

In Homekit, I get 3 Sensors (as motionsensors) but i‘m only able to activate the first sensor with gpio 16. 
the others don‘t show a „motion“ when I get 3.3v to the gpios 14 and 5. I hope you can help me. 
greetings
Martin

**Comment by pilot1981** [comment]
Please can you share your HW project?

**Comment by Hanimaniac** [comment]
Hello Pilot,
I have a wintergarden (in german Wintergarten) at my house, which is climated and uses blinds on the roof. Its own „automation“ works with this wind/ rain/ and lightsensors. My Idea is to use the signals from the sensors (12v) and bring the signals with a transistor to the 3.3v gpios on the wemos.

**Comment by Hanimaniac** [comment]
So I think I found a solution for my Problem.

Like the issue in #1729 I had the same strange behavior with a wemos D1 mini. 
So I took a [Shelly i3](https://shelly.cloud/knowledge-base/devices/shelly-i3/), which is a smart switch with 3 Inputs.
After flashing with HAA and this JSON:
 {"c":{"o":1},"a":[{"t":12,"f1":[{"g":14,"p":0}],"s":0,"i":5},{"t":12,"s":0,"i":5,"f1":[{"g":12,"p":0}]},{"t":12,"f1":[{"g":13,"p":0}],"s":0,"i":5}]}]
I had three working motion sensors for my Project. The only Problem was, that the i3 works with 230V AC or 24-60V DC.

What I needed was 12V. So I found this [tutorial](https://www.shelly-support.eu/forum/index.php?thread/11911-shelly-i3-stromversorgung-mit-5v-bis-12v-dc/) which describes how to get the shelly i3 work with 12V. 
I tried it and it works perfect.

Edit 23.02.2022
Like in my post #1752, I changed the json so that the Input to the motionsensors work like push buttons (which is the right way for my project)
json: 

{"c":{"o":1},"a":[{"t":12,"b":[{"g":14,"p":0},{"g":14,"t":0}],"s":0},{"t":12,"b":[{"g":12,"p":0},{"g":12,"t":0}],"s":0},{"t":12,"b":[{"g":13,"p":0},{"g":13,"t":0}],"s":0}]}

Greeting 
Hanimaniac

**Reply by sorriso93** [reply]
well then seems something related to hardware on wemos d1 mini... unfortunately I only use that hardware to build up my devices
The difference I see between your hardware and my configuration is the conversion from 12v to 3.3v to the input of wemos, this is not my case becouse I use standard wemos shield...

**Reply by Hanimaniac** [reply]
You're right,
I also cannot understand, why there is this strange behavior on the d1 mini. I use the wemos in like 85% in my home automation, with no problems at all.
The good thing on the shelly i3 is, that it is just slightly bigger than the wemos.

**Comment by Kienz** [comment]
@Hanimaniac Which wind sensor did you use?

**Reply by Hanimaniac** [reply]
Hello @Kienz , 
its a standard windsensor like this: https://amzn.eu/d/b11vTjQ
I used the old controller of my wintergarden for the evaluation. Unfortunately, it looks like something there is broke, so now i look for a new evaluate unit. May when you look at "Grenzwertschalter Wind", you can find somethink that works for you.
